lib/compilers/glimmer/index.js

```diff
@@ -12,13 +12,28 @@ export default class GlimmerTemplatePrecompiler extends Filter {
   }
 
   processString(content, relativePath) {
-    let specifier = getTemplateSpecifier(this.rootName, relativePath);
-    return `export default ${this.precompile(content, { meta: { specifier, '<template-meta>': true } })};`;
+    let specifier = getTemplateSpecifier(this.rootName, removeManagerIdFromPath(relativePath));
+    let managerId = getManagerIdFromPath(relativePath);
+    let meta = { specifier };
+
+    if (managerId) {
+      meta.managerId = managerId;
+    }
+
+    return `export default ${this.precompile(content, { meta })};`;
   }
 
   precompile(content, options) {
     return precompile(content, options);
   }
+
+  getDestFilePath(relativePath) {
+    let processedPath = super.getDestFilePath(relativePath);
+
+    if (!processedPath) return processedPath;
+
+    return removeManagerIdFromPath(processedPath);
```

Comment:
The module name is the full path on disk relative to the project root and excluding the extension. In general (e.g. in a normal ember app), you can absolutely have modules that have a `.` in them.